test/unit/settings_test.py

```diff
@@ -133,21 +132,21 @@ def test_860m(self):
 class TestEiaSettings:
     """Test pydantic model that validates EIA datasets."""
 
-    def test_eia923_dependency(self):
-        """Test 860 is added if 923 is specified and 860 is not."""
-        eia923_settings = Eia923Settings()
-        settings = EiaSettings(eia923=eia923_settings)
-        data_source = DataSource.from_id("eia860")
+    # def test_eia923_dependency(self):
+    #     """Test 860 is added if 923 is specified and 860 is not."""
+    #     eia923_settings = Eia923Settings()
+    #     settings = EiaSettings(eia923=eia923_settings)
+    #     data_source = DataSource.from_id("eia860")
 
-        assert settings.eia860
+    #     assert settings.eia860
 
-        assert settings.eia860.years == data_source.working_partitions["years"]
+    #     assert settings.eia860.years == data_source.working_partitions["years"]
 
-    def test_eia860_dependency(self):
-        """Test 923 tables are added to eia860 if 923 is not specified."""
-        eia860_settings = Eia860Settings()
-        settings = EiaSettings(eia860=eia860_settings)
-        assert settings.eia923.years == eia860_settings.years
+    # def test_eia860_dependency(self):
+    #     """Test 923 tables are added to eia860 if 923 is not specified."""
+    #     eia860_settings = Eia860Settings()
+    #     settings = EiaSettings(eia860=eia860_settings)
+    #     assert settings.eia923.years == eia860_settings.years
```

Comment:
we chatted about this on the ☎️  and came to the understanding that despite this restriction not being technically necessary across the board, it'd be good to require an overlap between 923 and 860 years.

Reply:
I think both of these tests could be converted to use `isdisjoint`. We want to make sure they are not disjoint, so i think we cold do something like this for both tests:

```
# assign both EIA form years
eia860_years = settings.eia860.years
eia923_years = data_source.working_partitions["years"]
# assert that there is some overlap between EIA years
assert not set(eia860_years).isdisjoint(eia923_years)
```

Reply:
I'm not sure what the difference is between these two tests? Did you mean to reverse the 860 and the 923 labels? Also, what is the difference between `eia923_years = data_source.working_partitions["years"]` and `eia923_years = settings.eia923.years`?

Reply:
they just some from different places: the working partitions comes from the stored working partitions while the `settings.eia923.years` come from the settings file that the test is checking.

Reply:
I guess I don't understand what you mean by 1st and 2nd test then. They seem to be doing the same thing?

Reply:
i believe these are both checked because of the way the settings classes themselves are constructed. in the `pudl.settings.EiaSettings.check_eia_dependencies` we are setting default values for 860 and 923 years which can either come from the settings file or is by default grabbed from the working partitions,

Reply:
Should we also test this in the reverse as in: 

```
set(eia923_years).isdisjoint(eia860_years)
```